doh-client/doh-client.conf

```diff
@@ -7,40 +7,52 @@ listen = [
 ]
 
 # HTTP path for upstream resolver
-# If multiple servers are specified, a random one will be chosen each time.
-upstream_google = [
 
-    # Google's productive resolver, good ECS, bad DNSSEC
-    "https://dns.google.com/resolve",
+# CloudFlare's resolver for Tor, available only with Tor
+# Remember to disable ECS below when using Tor!
+# Blog: https://blog.cloudflare.com/welcome-hidden-resolver/
+#"https://dns4torpnlfs2ifuz2s2yf3fc7rdmsbhm6rw75euj35pac6ap25zgqad.onion/dns-query",
 
-    # CloudFlare's resolver, bad ECS, good DNSSEC
-    #"https://cloudflare-dns.com/dns-query",
-    #"https://1.1.1.1/dns-query",
-    #"https://1.0.0.1/dns-query",
 
-    # CloudFlare's resolver for Tor, available only with Tor
-    # Remember to disable ECS below when using Tor!
-    # Blog: https://blog.cloudflare.com/welcome-hidden-resolver/
-    #"https://dns4torpnlfs2ifuz2s2yf3fc7rdmsbhm6rw75euj35pac6ap25zgqad.onion/dns-query",
+[upstream]
 
-]
-upstream_ietf = [
+# available selector: random or weighted_round_robin
+upstream_selector = "random"
 
-    # Google's experimental resolver, good ECS, good DNSSEC
-    #"https://dns.google.com/experimental",
+# weight should in (0, 100], if upstream_selector is random, weight will be ignored
 
-    # CloudFlare's resolver, bad ECS, good DNSSEC
-    #"https://cloudflare-dns.com/dns-query",
-    #"https://1.1.1.1/dns-query",
-    #"https://1.0.0.1/dns-query",
+# Google's productive resolver, good ECS, bad DNSSEC
+[[upstream.upstream_google]]
+url = "https://dns.google.com/resolve"
+weight = 50
 
-    # CloudFlare's resolver for Tor, available only with Tor
-    # Remember to disable ECS below when using Tor!
-    # Blog: https://blog.cloudflare.com/welcome-hidden-resolver/
-    #"https://dns4torpnlfs2ifuz2s2yf3fc7rdmsbhm6rw75euj35pac6ap25zgqad.onion/dns-query",
+# CloudFlare's resolver, bad ECS, good DNSSEC
+[[upstream.upstream_google]]
+url = "https://cloudflare-dns.com/dns-query"
+weight = 50
+
+# CloudFlare's resolver, bad ECS, good DNSSEC
+[[upstream.upstream_google]]
+url = "https://1.1.1.1/dns-query"
+weight = 50
+
+# Google's experimental resolver, good ECS, good DNSSEC
+[[upstream.upstream_ietf]]
+url = "https://dns.google.com/experimental"
+weight= 50
```

Comment:
Please add a space here and on the following lines.

Reply:
Add space now